### PR TITLE
379/endpoint removing connections

### PR DESCRIPTION
**Pull request by HGodal**
Open for suggestions/improvements

The new endpoint `admin/connection?location=...&rowKey=...` takes location and rowkey of a giver as input, and deletes the corresponding connection. This includes both completed connections and connections waiting to be confirmed.

### CODE REVIEW

File: GiEnJul/Controllers/AdminController.cs [hunk 1]
```diff
@@ -132,6 +132,58 @@ public async Task<ActionResult> PutWishAsync(string person_rowkey, [FromBody] st
...
 
+        [HttpDelete("Connection")]
+        [Authorize(Policy = "DeleteConnection")]
+        public async Task<ActionResult> DeleteConnectionAsync(string location, string rowKey)
+        {
+            var giver = await _giverRepository.GetGiverAsync(location, rowKey);
+
+            if (giver?.MatchedRecipient is null)
+            {
+                return NoContent();
```
Comment (Sommerli) [inline GiEnJul/Controllers/AdminController.cs:143]: I belive `return NotFound();` would be more fitting here

https://docs.microsoft.com/en-us/aspnet/core/web-api/action-return-types?view=aspnetcore-5.0#synchronous-action-1

File: GiEnJul/Controllers/AdminController.cs [hunk 2]
```diff
@@ -132,6 +132,58 @@ public async Task<ActionResult> PutWishAsync(string person_rowkey, [FromBody] st
...
+            }
+
+            var recipient = await _recipientRepository.GetRecipientAsync(location, giver.MatchedRecipient);
+
```
Comment (Sommerli) [inline GiEnJul/Controllers/AdminController.cs:147]: Could also do a null check on recipient here

File: GiEnJul/Controllers/AdminController.cs [hunk 3]
```diff
@@ -132,6 +132,58 @@ public async Task<ActionResult> PutWishAsync(string person_rowkey, [FromBody] st
...
+            {
+                await _giverRepository.InsertOrReplaceAsync(giver);
+                await _recipientRepository.InsertOrReplaceAsync(recipient);
+
+                if (await _connectionRepository.ConnectionExists(giver, recipient))
+                {
+                    await _connectionRepository.DeleteConnectionAsync(location, recipient.RowKey + "_" + giver.RowKey);
+                }
+            }
+            catch (NullReferenceException)
```
Comment (Sommerli) [inline GiEnJul/Controllers/AdminController.cs:169]: The operations in the try block could throw other exceptions than NullReferenceException. Should another exception be thrown, the tables might get faulty data as not all operations are completed.  I suggest you change from NullReferenceException to Exception e, and do a _log.Error call.

File: GiEnJul/Controllers/AdminController.cs (hunk 2 above)
Comment (HGodal) [reply GiEnJul/Controllers/AdminController.cs:147]: Will do

File: GiEnJul/Controllers/AdminController.cs (hunk 3 above)
Comment (HGodal) [reply GiEnJul/Controllers/AdminController.cs:169]: 👍🏼

**Review by TTorgersen — APPROVED** [review]

File: GiEnJul/Controllers/AdminController.cs [hunk 4]
```diff
@@ -132,6 +132,56 @@ public async Task<ActionResult> PutWishAsync(string person_rowkey, [FromBody] st
...
+            recipient.HasConfirmedMatch = false;
+            recipient.IsSuggestedMatch = false;
+            recipient.MatchedGiver = null;
+
+            try
+            {
+                await _giverRepository.InsertOrReplaceAsync(giver);
+                await _recipientRepository.InsertOrReplaceAsync(recipient);
+
+                if (await _connectionRepository.ConnectionExists(giver, recipient))
```
Comment (Sommerli) [inline GiEnJul/Controllers/AdminController.cs]: This if and contents should be put outside the try catch. Should the if be false, the giver and recipient while no connection is updated.

The if and its contents should probably be put before the try catch.

**Review by Sommerli — APPROVED** [review]

**Pull request merged**